provider/metrics/metrics.go

```diff
@@ -0,0 +1,223 @@
+package metrics
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/patrickmn/go-cache"
+	"github.com/pkg/errors"
+	"github.com/virtual-kubelet/azure-aci/client/aci"
+	"github.com/virtual-kubelet/virtual-kubelet/log"
+	stats "github.com/virtual-kubelet/virtual-kubelet/node/api/statsv1alpha1"
+	"github.com/virtual-kubelet/virtual-kubelet/trace"
+	"golang.org/x/sync/errgroup"
+	v1 "k8s.io/api/core/v1"
+)
+
+const (
+	ContainerGroupCacheTTLSeconds = 60 * 5
+)
+
+// package dependency: query the Pods in current virtual nodes. it usually is ResourceManager
+type PodGetter interface {
+	GetPods() []*v1.Pod
+}
+
+// package dependency: query the Pod's correspoinding Container Group metrics from Container Insights
+type ContainerGroupMetricsGetter interface {
+	GetContainerGroupMetrics(ctx context.Context, resourceGroup, containerGroup string, options aci.MetricsRequest) (*aci.ContainerGroupMetricsResult, error)
```

Comment:
`GetMetrics`

Reply:
This name is to match with the implementation one in client\aci\metrics.go.  The purpose of the Interface here just for mocking in unit test.

Reply:
I see.
Perhaps we are abstracting the wrong thing here?
We are exporting an interface so that we can provide a mocked implementation for unit tests for something that otherwise only has a single implementation.

Don't want to hold up the PR on such semantics.